07a_system_activity_plots.R

```diff
@@ -1,283 +1,264 @@
-  "Exited to \nNon-Permanent Destination"
-)
-
-system_activity_prep <- reactive({
-  sys_inflow_outflow_plot_data()() %>% # this is a people-level df
-    filter(InflowTypeDetail != "something's wrong" &
-             OutflowTypeDetail != "something's wrong") %>%
-    pivot_longer(
-      cols = c(InflowTypeDetail, OutflowTypeDetail), 
-      names_to = "x.axis.var", 
-      values_to = "cat.var") %>%
-    group_by(x.axis.var, cat.var) %>%
+system_activity_prep_detail <- reactive({
+# browser()
+  inflow <- sys_inflow_outflow_plot_data()() %>%
+    select(PersonalID,
+           InflowTypeSummary,
+           InflowTypeDetail) %>%
+    group_by(InflowTypeDetail) %>%
     summarise(values = n()) %>%
-    # filter(!is.na(cat.var)) %>%
-    mutate(
-      values = ifelse(x.axis.var == "OutflowTypeDetail", values * -1, values),
-      inflow_outflow = x.axis.var,
-      x.axis.var = case_when(
-        x.axis.var == "InflowTypeDetail" &
-          cat.var %in% active_at_vals
-        ~ active_as_of_start(),
-        
-        x.axis.var == "OutflowTypeDetail" &
-          cat.var %in% active_at_vals
-        ~ active_as_of_end(),
-          
-        x.axis.var == "InflowTypeDetail"
-        ~ "Inflow",
-        
-        x.axis.var == "OutflowTypeDetail"
-        ~ "Outflow"
-      )
-    )
-})
-
-# Combine the inflow types (newly homeless, returned from permanent, re-engaged
-# into one group)
-system_activity_summary_prep <- reactive({
-  system_activity_prep() %>%
-    mutate(
-      cat.var = case_when(
-        x.axis.var == "Inflow" &
-        !(cat.var %in% active_at_vals)
-        ~ "Inflow",
+    ungroup() %>%
+    rename("Status" = InflowTypeDetail) %>%
+    full_join(frame_detail %>%
+                filter(InflowOutflow == "Inflow")) %>%
+    mutate(values = replace_na(values, 0))
   
-        x.axis.var == "Outflow" &
-        !(cat.var %in% active_at_vals)
-        ~ "Outflow",
+  outflow <- sys_inflow_outflow_plot_data()() %>%
+    select(PersonalID,
+           OutflowTypeSummary,
+           OutflowTypeDetail) %>%
+    group_by(OutflowTypeDetail) %>%
+    summarise(values = n()) %>%
+    ungroup() %>%
+    rename("Status" = OutflowTypeDetail) %>%
+    full_join(frame_detail %>%
+                filter(InflowOutflow == "Outflow")) %>%
+    mutate(values = replace_na(values, 0))
   
-        TRUE ~ cat.var
+  inflow %>%
+    full_join(outflow, join_by(Time, values, Status, InflowOutflow)) %>%
+    mutate(
+      Time = factor(
+        Time,
+        levels = c("Active at Start",
+                   "Inflow",
+                   "Outflow",
+                   "Active at End")
+      ),
+      Status = factor(
+        Status,
+        levels = c(
+          "Homeless",                          
+          "Housed",
+          "Newly Homeless",
+          "Returned from \nPermanent",
+          "Re-engaged from \nNon-Permanent",
+          "Exited to \nNon-Permanent Destination",
+          "Exited to \nPermanent Destination",
```

Comment:
i think we decided to shorten this to "Exited, Permanent" to match the sankey chart and to shorten the label